samples/FleetProvisioning/README.md

```diff
@@ -306,7 +319,144 @@ aws iot create-provisioning-claim \
 Finally, you can also pass the certificate signing request while invoking the Fleet Provisioning sample.
 
 ``` sh
-mvn compile exec:java -pl samples/FleetProvisioning -Dexec.mainClass="fleetprovisioning.FleetProvisioningSample" -Dexec.args="--endpoint <endpoint>
+mvn compile exec:java -pl samples/FleetProvisioning -Dexec.mainClass="fleetprovisioning.Mqtt5FleetProvisioningSample" -Dexec.args="--endpoint <endpoint>
 --cert <path to certificate> --key <path to private key> --template_name <template name> --template_parameters '{\"SerialNumber\":\"1\",\"DeviceLocation\":\"Seattle\"}'  --csr <path to csr file>"
 ```
 
+## Service Client Notes
+### Difference relative to MQTT311 IotIdentityClient (Fleet Provisioning)
+The IotIdentityClient with mqtt5 client is almost identical to the mqtt3 one. We wrapped the Mqtt5Client into MqttClientConnection so that we could keep the same interface for IotIdentityClient.
```

Comment:
Trivial: mqtt is capitalized elsewhere as "Mqtt", unsure if we need to stick with that here for "mqtt5" and "mqtt3".

I'll get to the IotIdentityClient during this review but leaving a note here that if the API and functionality for the customer is identical, we may not want to say "almost identical" because that might confuse them into thinking something is different about the client. We can change this to something like "The IotIdentityClient provides identical functionality whether it is built using an Mqtt5 client or an Mqtt311 Client Connection." Unless the customer has to engage with the MQTT5toMQTT3Adapter themselves, might be better to just leave it out of the conversation entirely.

Reply:
Nvm, Just got down to the section below. I thought we were going to allow creation of service clients by providing the mqtt5 client directly. Mix up in communication. Yea, let's leave the info about wrapping the Mqtt5 Client into an MqttClientConnection.

That does make it a bit more confusing for the customer but they have instructions here so it should be fine.

Reply:
In other SDKs, we can create service clients by providing the Mqtt5 client directly. By passing the mqtt5 client, the service client would create a MqttClientConnection from Mqtt5 client. However, for Java, If the service client is in charge of creation of the connection, it should call `connection.close()` to release the MqttClientConnection on shutdown. Unfortunately, there is no way for a service client to call a `connection.close()` on shutdown without introduce a new API. 
As a compromise, we have to ask the user to manage the MqttClientConnection resource.